packages/client/src/components/Input/FormInput.component.js

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import validateForm from '../../utils/validateForm';
+import './FormInput.style.css';
+
+const FormInput = ({ value, type, name, label, placeholder, onChange }) => {
+  const [errors, setErrors] = useState({});
+
+  const handleChange = (event) => {
+    setErrors(validateForm(event.target.value));
+    onChange(value);
+  };
```

Comment:
This is kind of a loop, isn't it?
When is the input value ever updated? 
How will you get the value in the component or container that will use the input - to know what to send to the database, f.x.? 
This code, on change, just runs the handle change, but nothing really happens and the input now is no typable. 
I will not go more specific for now, just let me know if it does not make sense and if you don't know what and how to change. 

You are very close though, good job! 💪🏽

Reply:
i think I need help. I am thinking to use value as a state instead of props. but the mentors suggested it better to use them as props so...

Reply:
yes, you should use it as a prop, and then the state update is also a prop actually! 

```js
// in the parent, when you will want to use the component:
 const [value, setValue] = useState("");
// different names though, as you will have several different inputs. Like name, setName etc. 
...
// the component gets at least one more prop, the state update:
const FormInput = ({ value, setValue, type, name, label, placeholder, onChange }) => { ... 
...
  const handleChange = (event) => {
    setErrors(validateForm(event.target.value));
    setValue(event.target.value);
  };
...
< FormInput value={name} setValue={setName} ....

```

Does this make sense?

Reply:
@Hailegebriel22 ☝🏽

Reply:
Yes, Thank you so much!